department-of-veterans-affairs/caseflow
# Appeal-19613: Development and Domo Seeds and Test data

…caseflow VhaCamo

Resolves 

- [Appeal-19614](https://vajira.max.gov/browse/APPEALS-19614)
- [Appeal-19615](https://vajira.max.gov/browse/APPEALS-19615)
- [Appeal-19616](https://vajira.max.gov/browse/APPEALS-19616)
- [19623](https://vajira.max.gov/browse/APPEALS-19623)
- [19624](https://vajira.max.gov/browse/APPEALS-19624)
- [19625](https://vajira.max.gov/browse/APPEALS-19628)
- [19627](https://vajira.max.gov/browse/APPEALS-19627)
- [19628](https://vajira.max.gov/browse/APPEALS-19628)
### Description
**Added Seeds for populating VhaCamo Organization and VhaCaregiver  queue to have 5 of each:** 
  - Assigned
  - In Progress
  - Completed
- added 3 new methods to seeds/tasks.rb 
- added 1 new method to factories/task.rb
- 
**Adding Seeds for VHA Program Offices**
  - Assigned
  - In Progress
  - Ready for Review
  - On Hold
  - Completed

**List of VHA Program Offices:**
  - Community Care - Payment Operations Management
  - Community Care - Veteran and Family Members Program
  - Member Services - Health Eligibility Center
  - Member Services - Beneficiary Travel
  - Prosthetics

- added 2 new methods to seeds/tasks.rb 
- added 1 new method to factories/task.rb

### Acceptance Criteria
- [ ] vha-camo should have pre-populated queues in all 3 tabs when switched to organization view.
- [ ] vha-caregiver should have pre-populated queue in all 3 tabs when switched to organization view.
### Testing Plan
1. Switch user to be either CAMOUser or CAMOADMIN 
2.  Go to /organizations/vha-camo
3.  All three tabs should have task prepopulated.
4.  `bundle exec rails runner scripts/run_seed_data_file.rb db/seeds/users.rb` -> this might cause an error since users are already in the db, you can either run all the seeds or comment out certain potion and just run this script to test.
5. ` bundle exec rails runner scripts/run_seed_data_file.rb db/seeds/veterans_health_administration.rb`
6. When clicking on the Case Detail link all task should be redirected to the detail page. 

![image](https://user-images.githubusercontent.com/122557351/229832291-91c36bc2-ff59-4c5c-bf1a-d3e947235e3d.png)


### User Facing Changes

 BEFORE|AFTER
 ---|---

### Code Documentation Updates

### Storybook Story
*For Frontend (Presentationa) Components*
